internal/status/route_status.go

```diff
@@ -306,15 +310,19 @@ func updateRouteObject(mRoute *routev1.Route, updateOption UpdateStatusOptions,
 }
 
 func compareRouteStatus(oldStatus, newStatus []routev1.RouteIngress) bool {
-	if len(oldStatus) != len(newStatus) {
+
+	if len(oldStatus) == 0 || len(newStatus) == 0 {
```

Comment:
In openshift, route may already have a status set by the default ingress controller, empty length check may not work. 
Is there a problem if we retain the old condition check ?

Reply:
@monotosh-avi  With addition of timestamp into status message, each time length differ even if there is no ip update (in old status message retrieved .. time comes with some modification.. due to which length differ). That leads to unnecessary updates and patching of routes. In my testing, I have seen len(newStatus) == 0 condition. len(oldStatus) condition is added for same reason (never encountered this during testing.. but added to take into account extreme case) . IF we are sure that there will be no case with len(oldStatus) == 0, we can take out len(oldStatus) == 0 condition.